opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -0,0 +1,1462 @@
+  Copyright 2014, 2015 Dr. Blatt - HPC-Simulation-Software & Services
+  Copyright 2014, 2015 Statoil ASA.
+  Copyright 2015 NTNU
+  Copyright 2015 IRIS AS
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_BLACKOILMODELEBOS_HEADER_INCLUDED
+#define OPM_BLACKOILMODELEBOS_HEADER_INCLUDED
+
+#include <ebos/eclproblem.hh>
+#include <ewoms/common/start.hh>
+
+#include <opm/autodiff/BlackoilModelParameters.hpp>
+#include <opm/autodiff/StandardWellsDense.hpp>
+#include <opm/autodiff/AutoDiffBlock.hpp>
+#include <opm/autodiff/AutoDiffHelpers.hpp>
+#include <opm/autodiff/GridHelpers.hpp>
+#include <opm/autodiff/WellHelpers.hpp>
+#include <opm/autodiff/BlackoilPropsAdInterface.hpp>
+#include <opm/autodiff/GeoProps.hpp>
+#include <opm/autodiff/WellDensitySegmented.hpp>
+#include <opm/autodiff/VFPProperties.hpp>
+#include <opm/autodiff/VFPProdProperties.hpp>
+#include <opm/autodiff/VFPInjProperties.hpp>
+#include <opm/autodiff/DefaultBlackoilSolutionState.hpp>
+#include <opm/autodiff/BlackoilDetails.hpp>
+#include <opm/autodiff/BlackoilModelEnums.hpp>
+#include <opm/autodiff/NewtonIterationBlackoilInterface.hpp>
+
+#include <opm/core/grid.h>
+#include <opm/core/linalg/LinearSolverInterface.hpp>
+#include <opm/core/linalg/ParallelIstlInformation.hpp>
+#include <opm/core/props/rock/RockCompressibility.hpp>
+#include <opm/common/ErrorMacros.hpp>
+#include <opm/common/Exceptions.hpp>
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/parser/eclipse/Units/Units.hpp>
+#include <opm/core/well_controls.h>
+#include <opm/core/simulator/SimulatorReport.hpp>
+#include <opm/core/simulator/SimulatorTimer.hpp>
+#include <opm/core/utility/parameters/ParameterGroup.hpp>
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/TableManager.hpp>
+
+#include <opm/autodiff/ISTLSolver.hpp>
+#include <opm/common/data/SimulationDataContainer.hpp>
+
+#include <dune/istl/owneroverlapcopy.hh>
+#include <dune/common/parallel/collectivecommunication.hh>
+
+#include <cassert>
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+#include <limits>
+#include <vector>
+#include <algorithm>
+//#include <fstream>
+
+
+
+namespace Ewoms {
+namespace Properties {
+NEW_TYPE_TAG(EclFlowProblem, INHERITS_FROM(BlackOilModel, EclBaseProblem));
```

Comment:
This brings back some not so nice memories of myself having a hard time understanding this property system and the template errors you get if something goes wrong. But hopefully nobody has to directly deal with it.